plugin/servergen.go

```diff
@@ -136,6 +136,20 @@ func (p *OrmPlugin) generateDefaultServer(file *generator.FileDescriptor) {
 	}
 }
 
+func (p *OrmPlugin) spanAnnotate(service autogenService, key, value, message string) {
+	withSpan := getServiceOptions(service.ServiceDescriptorProto).WithSpan
+	if withSpan != nil && *withSpan {
+		rawVarName := fmt.Sprint("rawParameter", strings.Title(key))
+		errVarName := fmt.Sprint("errMarshaling", strings.Title(key))
+		p.P(rawVarName, `, `, errVarName, ` := `, p.Import(encodingJsonImport), `.Marshal(`, value, `)`)
+		p.P(`if `, errVarName, ` != nil {`)
+		p.P(`span.Annotate([]`, p.Import(ocTraceImport), `.Attribute{`, p.Import(ocTraceImport), `.StringAttribute("`, key, `", "")}, "`, message, `")`)
+		p.P(`return nil, `, errVarName)
+		p.P(`}`)
+		p.P(`span.Annotate([]`, p.Import(ocTraceImport), `.Attribute{`, p.Import(ocTraceImport), `.StringAttribute("`, key, `", string(`, rawVarName, `))}, "`, message, `")`)
```

Comment:
Should we instead of generating this piece of code twice for each repository generate generalised function once and use it in all methods?

Reply:
Refactoring should be fine, provided you add the correct error handling code for the span.